app/src/main/java/me/phh/treble/app/Transsion.kt

```diff
@@ -0,0 +1,30 @@
+package me.phh.treble.app
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.preference.PreferenceManager
+import android.util.Log
+import java.io.File
+import java.lang.Exception
+
+object Transsion: EntryStartup {
+    val spListener = SharedPreferences.OnSharedPreferenceChangeListener { sp, key ->
+        when(key) {
+            TranssionSettings.usbOtg -> {
+                val b = sp.getBoolean(key, false)
+                val value = if(b) "1" else "0"
+                Misc.safeSetprop("persist.sys.phh.transsion.usbotg", value)
+            }
+        }
+    }
+
+    override fun startup(ctxt: Context) {
+        if(!TranssionSettings.enabled()) return
+        Log.d("PHH", "Starting Transsion service")
+        val sp = PreferenceManager.getDefaultSharedPreferences(ctxt)
+        sp.registerOnSharedPreferenceChangeListener(spListener)
+
+        //Refresh parameters on boot
+        spListener.onSharedPreferenceChanged(sp, TranssionSettings.usbOtg)
```

Comment:
Remove this so that the shell script is not called for everyone on every boot, just for those who set it